h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostUpdater.java

```diff
@@ -37,6 +37,12 @@ private XGBoostUpdater(Key<XGBoostModel> modelKey, DMatrix trainMat, BoosterParm
     _rabitEnv = rabitEnv;
     _in = new SynchronousQueue<>();
     _out = new SynchronousQueue<>();
+    setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
```

Comment:
This is the exact same fix I did, isn't it? :)

The only difference is that I am calling it after the updater is instantiated because calling an overridable method in a constructor is generally not a good practice.

Reply:
I am not sure I follow, I have searched if setUncaughtExceptionHandler is used anywhere else and did not find it

I have replaced throwing the exception with logging since it does not appear anywhere and If there are any undeclared exceptions thrown the handler will log them as well

Reply:
I didn't merge it to rel-yates, only to master: https://github.com/h2oai/h2o-3/pull/3456/files#diff-60c0595923fbd4a973dfececd24bd72dR195

sorry, I will push the commit there too - you can change it to your liking

Reply:
I think the handler is still set 2x

Reply:
true, fixed